File: scripts/syntax_check_all.py

import os
import sys
import py_compile
from pathlib import Path

def check_syntax(directory):
    """Check syntax."""
    print(f"🔍 Starting Syntax Check in: {directory}")
    print("=" * 60)
    
    error_count = 0
    checked_count = 0
    
    for root, dirs, files in os.walk(directory):
        # Skip virtualenvs and git
        if ".venv" in root or ".git" in root or "__pycache__" in root:
            continue
            
        for file in files:
            if not file.endswith(".py"):
                continue
            checked_count += 1
            full_path = str(Path(root) / file)
            try:
                py_compile.compile(full_path, doraise=True)
            except py_compile.PyCompileError as e:
                print(f"\n❌ Syntax Error in: {full_path}")
                print(f"   {e}")
                error_count += 1
            except Exception as e:
                print(f"\n⚠️ Unexpected Error checking {full_path}: {e}")
                error_count += 1

    print("\n" + "=" * 60)
    if error_count == 0:
        print(f"✅ Success! Scanned {checked_count} files. No syntax errors found.")
        sys.exit(0)
    else:
        print(f"❌ Failed! Found {error_count} syntax errors in {checked_count} files.")
        sys.exit(1)

if __name__ == "__main__":
    # Scan from current working directory
    check_syntax(os.getcwd())

File: rust_output/src/syntax_check_all.rs
use anyhow::{Result, Context};
use std::path::PathBuf;

/// Check syntax.
pub fn check_syntax(directory: String) -> Result<()> {
    // Check syntax.
    println!("🔍 Starting Syntax Check in: {}", directory);
    println!("{}", ("=".to_string() * 60));
    let mut error_count = 0;
    let mut checked_count = 0;
    for (root, dirs, files) in os::walk(directory).iter() {
        if (root.contains(&".venv".to_string()) || root.contains(&".git".to_string()) || root.contains(&"__pycache__".to_string())) {
            continue;
        }
        for file in files.iter() {
            if !file.ends_with(&*".py".to_string()) {
                continue;
            }
            checked_count += 1;
            let mut full_path = (PathBuf::from(root) / file).to_string();
            // try:
            {
                py_compile.compile(full_path, /* doraise= */ true);
            }
            // except py_compile.PyCompileError as _e:
            // except Exception as _e:
        }
    }
    println!("{}", ("\n".to_string() + ("=".to_string() * 60)));
    if error_count == 0 {
        println!("✅ Success! Scanned {} files. No syntax errors found.", checked_count);
        std::process::exit(0);
    } else {
        println!("❌ Failed! Found {} syntax errors in {} files.", error_count, checked_count);
        std::process::exit(1);
    }
}
